### PR TITLE
Add taiga_issue module

### PR DESCRIPTION
**Pull request by lekum**
Add a module to create issues on the Taiga Project Management system (https://taiga.io/).

### CODE REVIEW

File: web_infrastructure/taiga_issue.py [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+  status:
+    description:
+      - The issue status. Must exist previously.
+    required: False
+    default: New
+  severity:
+    description:
+      - The issue severity. Must exist previously.
+    required: False
+    default: New
```
Comment (bcoca) [inline web_infrastructure/taiga_issue.py]: new severity? seems default should be 'Normal'

File: web_infrastructure/taiga_issue.py [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+
+DOCUMENTATION = '''
+---
+module: taiga_issue
+short_description: Creates/deletes an issue in a Taiga Project Management Platform
+description:
+  - Creates/deletes an issue in a Taiga Project Management Platform (U(https://taiga.io)).
+  - An issue is identified by the combination of project, issue subject and issue type.
+  - This module implements the creation or deletion of issues (not the update).
+version_added: "1.9"
```
Comment (bcoca) [inline web_infrastructure/taiga_issue.py]: this should be 2.0

File: web_infrastructure/taiga_issue.py (hunk 1 above)
Comment (lekum) [inline web_infrastructure/taiga_issue.py]: You are right, it is a typo in the DOCUMENTATION variable

**Comment by bcoca** [comment]
not sure if this belongs in web_infrastructure or notification. @ansible/core anyone care to weigh in?

**Comment by gregdek** [comment]
Errors in travis from elsewhere... looking into it.

**Comment by lekum** [comment]
Can I help somehow?

**Comment by abadger** [comment]
I think web_infrastructure.... Ticketing systems can be used for notification but they have additional functionality as well so it is probably better in the more general category (unless we want a new issue_trackers category).

**Comment by lekum** [comment]
The Travis fails seems unrelated to the module... do you want me to somehow push something again in order to re-run them?

**Comment by lekum** [comment]
I have pushed again and the tests have passed. How can I help now?

**Comment by robynbergeron** [comment]
Hi @lekum -- I think this is ready for community_review now. Some standardized information follows:

Thanks for submitting this new module to Ansible Extras! This module is now in community review, a process that is open to all Ansible users. In order for this module to be approved, it must gain the following votes:

“works_for_me”: If you have tested the module thoroughly, including testing of all of the module’s options, and if the module works for you, please add “works_for_me” in the comments.

“passes_guidelines”: If you have gone through the module guidelines and the module meets all of the requirements, please add “passes_guidelines” in the comments. Guidelines are available here: http://docs.ansible.com/developing_modules.html#module-checklist

“needs_revision”: If the module fails to work for you, or if it doesn’t meet guidelines, please add “needs_revision” in the comments with details about what needs to be fixed.

When a module has both “works_for_me” and “passes_guidelines” tags, we will promote the module for inclusion in Ansible Extras.  At this point, you will be expected to maintain the module by fixing bugs and evaluating pull requests in a timely manner. 

Thanks again for submitting your Ansible module!

**Comment by gregdek** [comment]
And @lekum -- the best way to help now is to find potential users of your module to test according to the above guidelines. When you find them, send them here and tell them to get to work. :)

**Comment by lekum** [comment]
In order to make the testing easier for the volunteers, I have created a repo with a README with an example playbook:

https://github.com/lekum/ee-ansible-with-python/tree/master/extending/module

Just create a [taiga.io](https://taiga.io/) account, clone the repo and follow the README instructions in the `extending/module` folder.

Thanks you very much!

**Comment by wkeeling** [comment]
Hi @lekum 

So I've had a run through various different testing scenarios against this module covering all of the options, on branch `stable-1.9`. I started with the happy path case using your example playbook.yml and then I tried changing the values of each of the different options. You'll be glad to know that I wasn't able to break it, and everything seemed to work fine. To summarize, this is what I did:
- Set up a Taiga account, created a new empty project, and then set my TAIGA_USERNAME and TAIGA_PASSWORD environment variables appropriately.
- Tried the happy path using the example playbook.yml to create/delete an issue in Taiga using values for each of the required options, in both a scrum and kanban project, and verified in Taiga that it worked correctly.
- Tested all types: Bug/Question/Enhancement
- Tested all severities: Wishlist/Minor/Normal/Important/Critical
- Tested all priorities: Low/Normal/High
- Tested different statuses: New/In progress/Ready for test/Closed/Needs Info/Rejected/Postponed
- Tried create an issue that already exists => "Issue already exists" message
- Tried to delete a non-existent issue => "Issue does not exist" message
- Tried create using a non-existent taiga_host => "NETWORK ERROR" message
- Tried create using a non-existent project => "Unable to find project" message  
- Tried create using a missing project => "missing required arguments: project" message
- Tried create using a missing subject => "missing required arguments: subject" message  
- Tried create using a missing issue_type => "missing required arguments: issue_type" message 
- Tried create using a missing priority => Issue created with "Normal" priority (default)
- Tried create using a missing status => Issue created with "New" status (default)  
- Tried create using a missing severity => Issue created with "Normal" severity (default) 
- Tried create using a missing description => Issue created with no description (Taiga defaults to "Empty space is so boring... go on be descriptive...")
- Tried create without attachment => Issue created with no attachment 
- Tried create using path to non-existent attachment file => "/path/to/non_existent.yml is not a file" message 
- Tried create using missing attachment description => Issue created with attachment and empty attachment message
- Tried create using missing tags => Issue created with no tags
- Tried invalid username (via environment variable) => "Username or password does not matches user" message
- Tried invalid password (via environment variable) => "Username or password does not matches user" message

So all of the above tests worked as I expected - in how they created/deleted issues in Taiga. The error paths also behaved as I would expect.

So this works_for_me

**Comment by bcoca** [comment]
@wkeeling nice testing

**Comment by lekum** [comment]
Thank you very much @wkeeling for your exhaustive testing :-)

**Pull request merged**